File: Examples/DotNET/SDK/CSharp/Protection/UnprotectDocument.cs
﻿//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2015 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Words.Cloud. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////

using System;
using Aspose.Cloud;

namespace CSharp.Protection
{
    class UnprotectDocument
    {
        public static void Run()
        {
            string inputfile = "doc-sample-unprotect.doc";
            string subdirection = "Protection";

            // Upload input file from local directory to Cloud Storage
            Common.UploadFile(inputfile, subdirection);

            // Unprotect a Word Document
            WordsProtectionRequest wordsProtectionRequest3 = new WordsProtectionRequest("aspose", string.Empty, ProtectionType.NoProtection);
            Common.WordsService.WordsDocumentProtection.UnprotectDocument(inputfile, Common.GetOutputFilePath(inputfile, true), wordsProtectionRequest3, Common.FOLDER);

            // Download output file from cloud storage and save on local directory
            string dataDir = Common.DownloadFile(inputfile, subdirection);

            Console.WriteLine("\nDocument unprotected successfully.\nFile saved at " + dataDir);
        }
    }
}

File: Examples/DotNET/SDK/VisualBasic/Protection/UnprotectDocument.vb
﻿Imports Aspose.Cloud

Namespace Protection
    Class UnprotectDocument
        Public Shared Sub Run()
            Dim inputfile As String = "doc-sample-unprotect.doc"
            Dim subdirection As String = "Protection"

            ' Upload input file from local directory to Cloud Storage
            Common.UploadFile(inputfile, subdirection)

            ' Unprotect a Word Document
            Dim wordsProtectionRequest3 As New WordsProtectionRequest("aspose", String.Empty, ProtectionType.NoProtection)
            Common.WordsService.WordsDocumentProtection.UnprotectDocument(inputfile, Common.GetOutputFilePath(inputfile, True), wordsProtectionRequest3, Common.FOLDER)

            ' Download output file from cloud storage and save on local directory
            Dim dataDir As String = Common.DownloadFile(inputfile, subdirection)

            Console.WriteLine(Convert.ToString(vbLf & "Document unprotected successfully." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace